source/server/listener_manager_impl.cc

```diff
@@ -354,12 +360,13 @@ void ListenerImpl::addFilterChainForServerNames(
 void ListenerImpl::addFilterChainForApplicationProtocols(
```

Comment:
Nit: Could you add `addFilterChainForSourceType()` instead of overloading `addFilterChainForApplicationProtocols()`?